packages/react/src/bags/hooks/useBagItem.ts

```diff
@@ -210,8 +99,18 @@ const useBagItem: UseBagItem = bagItemId => {
     }
 
     const quantityToAdd = quantityToHandle - bagItem.quantity;
+    const productSize = bagItem?.product?.sizes?.find(
```

Comment:
If we provide the `bagItem` to the hook I think we can remove this logic from here 😁

Reply:
Removed!